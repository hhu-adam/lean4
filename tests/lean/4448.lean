import Lean

@[deprecated]
theorem hi : True := .intro

example : True := by
  simp [hi, hi]
