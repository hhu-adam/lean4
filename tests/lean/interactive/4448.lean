@[deprecated]
theorem hi : True := .intro

set_option linter.deprecated false in
@[deprecated]
theorem hi_self : hi = hi := rfl

example : True := by
  simp [‚Üê hi_self, hi]
                   --^ collectDiagnostics
