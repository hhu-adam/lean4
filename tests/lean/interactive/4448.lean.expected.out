{"version": 1,
 "uri": "file:///4448.lean",
 "diagnostics":
 [{"tags": [2],
   "source": "Lean 4",
   "severity": 2,
   "range":
   {"start": {"line": 8, "character": 10}, "end": {"line": 8, "character": 17}},
   "message": "`hi_self` has been deprecated",
   "fullRange":
   {"start": {"line": 8, "character": 10},
    "end": {"line": 8, "character": 17}}},
  {"tags": [2],
   "source": "Lean 4",
   "severity": 2,
   "range":
   {"start": {"line": 8, "character": 19}, "end": {"line": 8, "character": 21}},
   "message": "`hi` has been deprecated",
   "fullRange":
   {"start": {"line": 8, "character": 19},
    "end": {"line": 8, "character": 21}}}]}
